### DISCUSSION TITLE
Locate returning different road information everytime

### DISCUSSION THREAD

**Discussion by girishyadav20** [valhalla/valhalla#4837] [Q&A]
Hi,

I am using Locate to identify road classification information for a lat/lon coordinate. When I execute locate for the same lat/lon values multiple time, the returned edge classification information is different everytime. Here is the code snippet:

```python
payload = {
  "verbose": True,
  "locations": [
    {
      "lon": lon,
      "lat": lat
    }
  ],
  "costing": "auto",
  "costing_options": {
    "auto": {
      "ignore_access": True,
      "ignore_restrictions": True
    }
  },
  "units": "miles"
}
locate_response = actor.locate(payload)
edges = locate_response[0]["edges"]
edge = edges[0]
road_type = edge.get("edge", {}).get("classification", {}).get("classification")
```

I am aware that, when multiple edge options are returned the order might be different in the edges list. But, I verified that the road classification is different even if a single edge is returned. I used default configuration.

Am I missing any configuration that is causing this?

Thanks!

**Comment by kevinkreiser** [comment]
just to rule out something in the python bindings, i would suggest you run the valhalla_service and try the `/locate` endpoint directly. if this also shows different results every time then you need to tell us what data you are using and give us a sample request or we cant help debug any further

**Comment by girishyadav20** [comment]
It turned out to be the issue within our Dataflow pipeline, NOT with the Valhalla Locate.
